EXILED/Exiled.API/Features/Toys/Speaker.cs

```diff
@@ -80,5 +102,186 @@ public float MinDistance
+
+        /// <summary>
+        /// Creates a new Speaker instance.
+        /// </summary>
+        /// <param name="controllerId">The controller ID for the SpeakerToy.</param>
+        /// <param name="position">The position to place the SpeakerToy.</param>
+        /// <param name="isSpatial">Indicates whether the audio is spatialized.</param>
+        /// <param name="spawn">Determines if the speaker should be spawned immediately.</param>
+        /// <returns>A new <see cref="Speaker"/> instance.</returns>
+        public static Speaker Create(byte controllerId, Vector3 position, bool isSpatial = false, bool spawn = true)
+        {
+            Speaker speaker = new(Object.Instantiate(ToysHelper.SpeakerBaseObject))
+            {
+                Position = position,
+                IsSpatial = isSpatial,
+                Base = { ControllerId = controllerId },
+                BroadcastTo = new List<Player>(),
+            };
+
+            if (spawn)
+                speaker.Spawn();
+
+            return speaker;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="Speaker"/> associated with a given <see cref="SpeakerToy"/>.
+        /// </summary>
+        /// <param name="speakerToy">The SpeakerToy instance.</param>
+        /// <returns>The corresponding Speaker instance.</returns>
+        public static Speaker Get(SpeakerToy speakerToy)
+        {
+            AdminToy adminToy = Map.Toys.FirstOrDefault(x => x.AdminToyBase == speakerToy);
+            return adminToy is not null ? adminToy as Speaker : new Speaker(speakerToy);
+        }
+
+        /// <summary>
+        /// Plays a single audio file through the speaker system. (No Arguments given (assuming you already preset those)).
+        /// </summary>
+        /// <param name="path">Path to the audio file to play.</param>
+        /// <param name="destroyAfter">Whether the Speaker gets destroyed after it's done playing.</param>
+        /// <returns>A boolean indicating if playback was successful.</returns>
+        public bool Play(string path, bool destroyAfter = false) => Play(path, Volume, MinDistance, MaxDistance, destroyAfter);
+
+        /// <summary>
+        /// Plays a single audio file through the speaker system.
+        /// </summary>
+        /// <param name="path">The file path of the audio file.</param>
+        /// <param name="volume">The desired playback volume. (0 to <see cref="float"/>) max limit.</param>
+        /// <param name="minDistance">The minimum distance at which the audio is audible.</param>
+        /// <param name="maxDistance">The maximum distance at which the audio is audible.</param>
+        /// <param name="destroyAfter">Whether the Speaker gets destroyed after it's done playing.</param>
+        /// <returns>A boolean indicating if playback was successful.</returns>
+        public bool Play(string path, float volume, float minDistance, float maxDistance, bool destroyAfter)
```

Comment:
It's not forced to be wrapped in an if? that's not required with a bool statement. You can do it without it.

Reply:
> If it's intended for people to ignore returned value then why return a value? Exceptions exist for a very specific purpose of discovering unexpected behaviour of a program. Having the wrong file path hardcoded or not having the audio is unexpected behaviour, not a normal part of program flow. Bool returns are usually used in this context to distinguish between 2 equally correct cases (Try pattern). Suppressing exceptions with them is just a bad practice.

It's intention is for in case someone wanted to make a command, and wants to know whether it was an invalid path or not. I can replace the Log.Warn with throwing an Exception, but other than that there is no point not to do this. (I'll still change this just trying to argue why I made it in the first place.)

Reply:
I don't get the point of this comment. Why function can't return value which indicates if play was successfull? This may be useful in future for some points. Also this value always can be omitted if necessary

Reply:
If you'll check exiled code you can notice that we throw exceptions rarely cuz usually we usually don't need to stop execution of caller method in case of fail. And as I said return value can be used in other cases